tests/lib/Factory/Export/ParametersFactoryTest.php

```diff
@@ -0,0 +1,265 @@
+            'languages' => 'eng-GB',
+            'page_size' => '500',
+            'web_hook' => null,
+            'host' => null,
+        ];
+
+        $this->siteAccessService
+            ->method('getAll')
+            ->willReturn(
+                [
+                    new SiteAccess($siteAccess),
+                ]
+            );
+
+        $this->credentialsResolver
+            ->expects(self::once())
+            ->method('hasCredentials')
+            ->with($siteAccess)
+            ->willReturn(true);
+
+        $this->getHostUriAndApiNotifierUri($siteAccess);
+
+        self::assertEquals(
+            Parameters::fromArray($this->options),
+            $this->parametersFactory->create($options, ParametersFactoryInterface::COMMAND_TYPE)
+        );
+    }
+
+    public function testCreateForMultiCustomerConfiguration(): void
+    {
+        $firstSiteAccess = 'test';
+        $secondSiteAccess = 'second_siteaccess';
+
+        $options = [
+            'customer_id' => '12345',
+            'license_key' => '12345-12345-12345-12345',
+            'siteaccess' => $firstSiteAccess,
+            'item_type_identifier_list' => 'article, product, blog',
+            'languages' => 'eng-GB',
+            'page_size' => '500',
+            'web_hook' => null,
+            'host' => null,
+        ];
+
+        $this->getAllSiteAccesses();
+
+        $this->credentialsResolver
+            ->expects(self::at(0))
+            ->method('hasCredentials')
+            ->with($firstSiteAccess)
+            ->willReturn(true);
+
+        $this->credentialsResolver
+            ->expects(self::at(1))
+            ->method('hasCredentials')
+            ->with($secondSiteAccess)
+            ->willReturn(true);
+
+        $this->getHostUriAndApiNotifierUri($firstSiteAccess);
+
+        self::assertEquals(
+            Parameters::fromArray($this->options),
+            $this->parametersFactory->create($options, ParametersFactoryInterface::COMMAND_TYPE)
+        );
+    }
+
+    /**
+     * @throws \EzSystems\EzRecommendationClient\Exception\MissingExportParameterException
+     */
+    public function testThrowExportCredentialsNotFoundException(): void
+    {
+        $siteAccess = 'invalid';
+
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage(sprintf('SiteAccess %s doesn\'t exists', $siteAccess));
+
+        $this->getAllSiteAccesses();
+
+        $this->parametersFactory->create(
+            [
+                'customer_id' => null,
+                'license_key' => null,
+                'siteaccess' => $siteAccess,
+                'item_type_identifier_list' => 'article, product, blog',
+                'languages' => 'eng-GB',
+                'page_size' => '500',
+                'web_hook' => null,
+                'host' => null,
+            ],
+            ParametersFactoryInterface::COMMAND_TYPE
+        );
+    }
+
+    private function getAllSiteAccesses(): void
```

Comment:
Method name doesn't match with what it does.

It configures site access service mock to return "all" site accesses.

Reply:
Fixed: https://github.com/ezsystems/ezrecommendation-client/pull/68/commits/6f0352a47a009acb0c8f5e11a407b0b2673ee726